File: CircularPath.cpp
#include "stdafx.h"
#include "CircularPath.h"
#include <math.h>


CircularPath::CircularPath(HWND hWnd, int window_width, int window_height)
{
	hWindow = hWnd; // save away window handle
	this->window_width = window_width; // save away window width
	this->window_height = window_height; // save away window width

	ball_colour = RGB(255, 0, 0);
	//circle coordinates
	a = 250;
	b = 265;
	
	x = a - r;
	y = b;

	// Pen and brush for travelling ball
	ball_pen = CreatePen(PS_SOLID, 1, RGB(0, 0, 0));
	ball_brush = CreateSolidBrush(ball_colour);

	hdc = GetDC(hWindow);

	SelectObject(hdc, ball_pen); // select ball pen colour
	SelectObject(hdc, ball_brush); // select ball brush colour
}

/*
* draws the ball/circle using the apt color
*/
void CircularPath::paint() {
	// draw a dot
	Ellipse(hdc, x, y, x + aWIDTH, y + aHEIGHT);
}

/*
* Repeatedly draws ball so as to simulate a continuous motion
*/
void CircularPath::moveCyclic() {
	// condition for continuing motion
	while (x <= a + r) {
		y = b - (int)round(sqrt(pow(r, 2) - pow((x - a), 2)));
		paint();

		y = b + (int)round(sqrt(pow(r, 2) - pow((x - a), 2)));
		paint();

		x += 20;
		// introduce a delay betweeen renderings
		Sleep(50);
	}
}


CircularPath::~CircularPath()
{
	DeleteObject(ball_pen);
	DeleteObject(ball_brush);

	ReleaseDC(hWindow, hdc);
}

File: CircularPath.java
package dymetric;

import java.awt.*;

public class CircularPath extends Canvas {

    protected Color ball_colour;
    //circle coordinates
    protected int a = 250;
    protected int b = 165;
    protected final int r = 150;
    protected int x = a - r;
    protected int y = b;
    protected final int aWIDTH, aHEIGHT;

    public CircularPath() {
        setBackground(Color.LIGHT_GRAY); // canvas color
        ball_colour = Color.RED;
        aWIDTH = aHEIGHT = 10;
    }

    // Feel free to double buffer if flickering occurs
    public void paint(Graphics g) {
        g.setColor(ball_colour);
        // draw a dot
        g.fillOval(x, y, aWIDTH, aHEIGHT);
    }

    public void moveCyclic() {
        // condition for continuing motion
        while (x <= a + r) {
            y = b - (int) Math.round(Math.sqrt(Math.pow(r, 2) - Math.pow((x - a), 2)));
            paint(this.getGraphics());

            y = b + (int) Math.round(Math.sqrt(Math.pow(r, 2) - Math.pow((x - a), 2)));
            paint(this.getGraphics());
            
            x += 20;
            // introduce a delay betweeen renderings
            try {
                Thread.sleep(50);
            } catch (InterruptedException e) {
            }
        }
    }
}
